### PR TITLE
Tidy from feedback:

### PR DESCRIPTION
**Pull request by johnoliver**
- Fix parsing Upstream versions
- Tidy up swagger doc

##### Checklist

- [x] `mvn clean package` build and test completes
- [x] documentation is changed or added

### CODE REVIEW

File: adoptopenjdk-api-v3-frontend/src/main/kotlin/net/adoptopenjdk/api/v3/routes/info/PlatformsResource.kt [hunk 1]
```diff
@@ -12,10 +13,12 @@ import javax.ws.rs.core.MediaType
 @Tag(name = "Release Info")
 @Path("info/platforms/")
 @Produces(MediaType.APPLICATION_JSON)
+@Schema(hidden = true)
```
Comment (karianna) [inline adoptopenjdk-api-v3-frontend/src/main/kotlin/net/adoptopenjdk/api/v3/routes/info/PlatformsResource.kt]: Document why you are hiding this.

File: adoptopenjdk-api-v3-frontend/src/main/kotlin/net/adoptopenjdk/api/v3/routes/info/VariantsResource.kt [hunk 2]
```diff
@@ -12,11 +14,13 @@ import javax.ws.rs.core.MediaType
 @Tag(name = "Release Info")
 @Path("info")
 @Produces(MediaType.APPLICATION_JSON)
+@Schema(hidden = true)
```
Comment (karianna) [inline adoptopenjdk-api-v3-frontend/src/main/kotlin/net/adoptopenjdk/api/v3/routes/info/VariantsResource.kt]: Doc this

File: adoptopenjdk-api-v3-frontend/src/main/kotlin/net/adoptopenjdk/api/v3/routes/info/VariantsResource.kt [hunk 3]
```diff
@@ -12,11 +14,13 @@ import javax.ws.rs.core.MediaType
...
 class VariantsResource {
 
     @GET
     @Path("variants")
-    @Operation(summary = "Returns information about available variants")
+    @Operation(summary = "Returns information about available variants", hidden = true)
```
Comment (karianna) [inline adoptopenjdk-api-v3-frontend/src/main/kotlin/net/adoptopenjdk/api/v3/routes/info/VariantsResource.kt]: Why hidden

File: adoptopenjdk-api-v3-models/src/main/kotlin/net/adoptopenjdk/api/v3/models/Asset.kt [hunk 4]
```diff
@@ -3,6 +3,7 @@ package net.adoptopenjdk.api.v3.models
 import org.eclipse.microprofile.openapi.annotations.media.Schema
 import java.time.LocalDateTime
 
+@Schema(hidden = true)
```
Comment (karianna) [inline adoptopenjdk-api-v3-models/src/main/kotlin/net/adoptopenjdk/api/v3/models/Asset.kt]: same again

File: adoptopenjdk-api-v3-models/src/main/kotlin/net/adoptopenjdk/api/v3/models/VersionData.kt [hunk 5]
```diff
@@ -8,26 +8,56 @@ class VersionData {
...
-        this.optional = optional
+        this.optional = if (optional?.isNotEmpty() == true) optional else null
         this.openjdk_version = openjdk_version
+        this.semver = formSemver()
+    }
+
+
+    // i.e 11.0.1+11.1
+    fun formSemver(): String? {
+        var semver = major.toString() + "." + minor + "." + security
```
Comment (karianna) [inline adoptopenjdk-api-v3-models/src/main/kotlin/net/adoptopenjdk/api/v3/models/VersionData.kt]: Is it really `security`? semver.org calls this `patch`, JEP 223 calls this `update` and the 4th number `patch`.

Pick one and say who's scheme you're following and say it's bullshit ;-)

File: adoptopenjdk-api-v3-models/src/main/kotlin/net/adoptopenjdk/api/v3/models/VersionData.kt [hunk 6]
```diff
@@ -8,26 +8,56 @@ class VersionData {
...
         this.build = build
-        this.optional = optional
+        this.optional = if (optional?.isNotEmpty() == true) optional else null
         this.openjdk_version = openjdk_version
+        this.semver = formSemver()
+    }
+
+
+    // i.e 11.0.1+11.1
+    fun formSemver(): String? {
```
Comment (karianna) [inline adoptopenjdk-api-v3-models/src/main/kotlin/net/adoptopenjdk/api/v3/models/VersionData.kt]: Test please?

File: adoptopenjdk-api-v3-persistance/src/main/kotlin/net/adoptopenjdk/api/v3/dataSources/APIDataStore.kt [hunk 7]
```diff
@@ -28,9 +28,10 @@ object APIDataStore {
         variants = JsonMapper.mapper.readValue(variantData, Variants::class.java)
 
         try {
-            loadDataFromDb()
+            binaryRepos = loadDataFromDb()
         } catch (e: Exception) {
-            LOGGER.error("Failed to read cache")
+            LOGGER.error("Failed to read db")
```
Comment (karianna) [inline adoptopenjdk-api-v3-persistance/src/main/kotlin/net/adoptopenjdk/api/v3/dataSources/APIDataStore.kt]: output `e` as well?

File: adoptopenjdk-api-v3-updater/src/main/kotlin/net/adoptopenjdk/api/v3/dataSources/github/VersionParser.kt [hunk 8]
```diff
@@ -61,7 +57,7 @@ class VersionParser {
 
     private fun matchAltPre223(versionString: String): Boolean {
```
Comment (karianna) [inline adoptopenjdk-api-v3-updater/src/main/kotlin/net/adoptopenjdk/api/v3/dataSources/github/VersionParser.kt]: Tests for the matchers?

File: adoptopenjdk-api-v3-updater/src/main/kotlin/net/adoptopenjdk/api/v3/mapping/upstream/UpstreamReleaseMapper.kt [hunk 9]
```diff
@@ -16,23 +16,36 @@ object UpstreamReleaseMapper : ReleaseMapper() {
...
-
         LOGGER.info("Getting binaries ${release_name}")
         val binaries = UpstreamBinaryMapper.toBinaryList(release.releaseAssets.assets)
         LOGGER.info("Done Getting binaries ${release_name}")
 
-        return Release(release.id, release_type, release_link, release_name, timestamp, updatedAt, binaries, download_count, vendor, versionData)
+        val versionData: VersionData
+
+        if (release_type == ReleaseType.ga && binaries.size > 0) {
+            //Release names for ga does not have full version name so take it from package
```
Comment (karianna) [inline adoptopenjdk-api-v3-updater/src/main/kotlin/net/adoptopenjdk/api/v3/mapping/upstream/UpstreamReleaseMapper.kt]: `does not` --> `do not have a full version name, so take it from the package`

File: adoptopenjdk-api-v3-models/src/main/kotlin/net/adoptopenjdk/api/v3/models/VersionData.kt (hunk 5 above)
Comment (johnoliver) [reply adoptopenjdk-api-v3-models/src/main/kotlin/net/adoptopenjdk/api/v3/models/VersionData.kt]: actually 223 calls this a security, can leave it

**Pull request merged**